### ISSUE TITLE
eslint-env doesn't work in single file as described in docs

### ISSUE THREAD

**Issue by Skriptach** [eslint/eslint#12262]
Tried to use
```
/* eslint-env es6, node */
```
and this
```
/* eslint-env es6, node, browser:false */
```

* **ESLint Version:**
v6.2.0
* **Node Version:**
v10.15.3
* **npm Version:**
6.9.0
* **yarn Version:**
1.16.0

**Please show your full configuration:**
```js
{
  "parser": "babel-eslint",
  "parserOptions": {
    "sourceType": "module",
    "allowImportExportEverywhere": true,
    "ecmaFeatures": {
      "legacyDecorators": true
    }
  },
  "extends": ["airbnb", "prettier", "prettier/flowtype", "prettier/react"],
  "env": {
    "browser": true,
    "es6": true,
    "node": true
  },
  "rules": {
    "arrow-parens": ["off"],
    "compat/compat": "error",
    "consistent-return": "off",
    "comma-dangle": "off",
    "indent": [
      "error",
      "tab",
      {
        "flatTernaryExpressions": true
      }
    ],
    "flowtype/boolean-style": ["error", "boolean"],
    "flowtype/define-flow-type": "warn",
    "flowtype/delimiter-dangle": ["error", "never"],
    "flowtype/generic-spacing": ["error", "never"],
    "flowtype/no-primitive-constructor-types": "error",
    "flowtype/no-weak-types": "warn",
    "flowtype/object-type-delimiter": ["error", "comma"],
    "flowtype/require-parameter-type": "off",
    "flowtype/require-return-type": "off",
    "flowtype/require-valid-file-annotation": "off",
    "flowtype/semi": ["error", "always"],
    "flowtype/space-after-type-colon": ["error", "always"],
    "flowtype/space-before-generic-bracket": ["error", "never"],
    "flowtype/space-before-type-colon": ["error", "never"],
    "flowtype/union-intersection-spacing": ["error", "always"],
    "flowtype/use-flow-type": "error",
    "flowtype/valid-syntax": "error",
    "generator-star-spacing": "off",
    "import/no-unresolved": "error",
    "import/no-extraneous-dependencies": "off",
    "import/prefer-default-export": "off",
    "jsx-a11y/anchor-is-valid": "off",
    "no-console": "off",
    "no-use-before-define": "off",
    "no-multi-assign": "off",
    "promise/param-names": "error",
    "promise/always-return": "error",
    "promise/catch-or-return": "error",
    "promise/no-native": "off",
    "react/sort-comp": [
      "error",
      {
        "order": [
          "type-annotations",
          "static-methods",
          "lifecycle",
          "everything-else",
          "render"
        ]
      }
    ],
    "react/jsx-no-bind": "off",
    "react/jsx-filename-extension": [
      "error",
      { "extensions": [".js", ".jsx"] }
    ],
    "react/jsx-props-no-spreading": "off",
    "react/prefer-stateless-function": "off",
    "react/static-property-placement": "off",
    "class-methods-use-this": "off"
  },
  "plugins": ["flowtype", "import", "promise", "compat", "react"],
  "settings": {
    "import/resolver": {
      "node": {
        "paths": ["app"]
      },
      "webpack": {
        "config": "configs/webpack.config.eslint.js"
      }
    }
  }
}
```

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

This is a 14th line in a file `app\main.dev.js`. But it is a most first invoked line. Previous are just comments.
```js
import { app, BrowserWindow, ipcMain, screen } from 'electron';
```

command in package.json
```js
{
  "scripts": {
    "lint": "cross-env NODE_ENV=development eslint --cache --format=pretty ."
  }
}
```

Get an error
```bash
  app\main.dev.js:14:39
  ×   14:39   screen is already declared in the upper scope.                 no-shadow
  ×  121:24   event is already declared in the upper scope.                  no-shadow
  ×  125:9    top is already declared in the upper scope.                    no-shadow
```

**Comment by mysticatea** [comment]
Hi.

I'm confused. The issue title mentions `eslint-env`, but the "What did you do?" section doesn't mention it. What is your problem?

**Comment by Skriptach** [comment]
Hi @mysticatea,
Sorry if described not clear.

I'm trying to switch off `browser` env for a single file `app\main.dev.js`.
But neither `/* eslint-env es6, node */` nor `/* eslint-env es6, node, browser:false */` worked when I add them as a 1st line

Thanks

**Comment by mysticatea** [comment]
Thank you for clarification.

Yes, `eslint-env` comment supports only adding envs.

You can configure it with [glob-based configuration](https://eslint.org/docs/user-guide/configuring#configuration-based-on-glob-patterns). For example:

```jsonc
{
    //...
    "overrides": [
        {
            "files": ["app/main.dev.js"],
            "env": { browser: false }
        }
    ]
}
```

**Comment by Skriptach** [comment]
@mysticatea,
Thank you very much for a quick response. That helped =)
